### PR TITLE
Add amgcl package

### PR DESCRIPTION
**Pull request by ImperatorS79**

### CODE REVIEW

File: mingw-w64-amgcl/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  fi
+
+  #Version of pastix too old in amgcl
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    "${MINGW_PREFIX}"/bin/cmake.exe \
+      -GNinja \
+      -DCMAKE_INSTALL_PREFIX="${MINGW_PREFIX}" \
+      -DAMGCL_BUILD_TESTS=ON \
+      -DAMGCL_BUILD_EXAMPLES=ON \
+      "${common_config[@]}" \
```
Comment (Biswa96) [inline mingw-w64-amgcl/PKGBUILD:47]: `common_config` is used here but `fortran_config` was declared first. Also add the `-DCMAKE_BUILD_TYPE` condition from PKGBUILD template.

File: mingw-w64-amgcl/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    fortran_config+=(
+      -DAMGCL_HAVE_FORTRAN=ON
+    )
+  fi
+
+  #Version of pastix too old in amgcl
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    "${MINGW_PREFIX}"/bin/cmake.exe \
+      -GNinja \
+      -DCMAKE_BUILD_TYPE=Release
```
Comment (mmuetzel) [inline mingw-w64-amgcl/PKGBUILD:44]: Should be:
``` diff
-      -DCMAKE_BUILD_TYPE=Release 
+      -DCMAKE_BUILD_TYPE=Release \
```

File: mingw-w64-amgcl/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+_realname=amgcl
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=1.4.3
+pkgrel=1
+pkgdesc="AMGCL is a header-only C++ library for solving large sparse linear systems with algebraic multigrid (AMG) method. (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
+url='http://amgcl.readthedocs.io/'
+license=('LICENSE')
```
Comment (mmuetzel) [inline mingw-w64-amgcl/PKGBUILD:12]: This should name the license for the package (ideally with a prefix `spdx:`). See, e.g., https://spdx.org/licenses/ for a list of license identifiers.

Comment (ImperatorS79) [reply mingw-w64-amgcl/PKGBUILD:12]: It is simply an MIT licence -> https://github.com/ddemidov/amgcl/blob/master/LICENSE.md.

**Comment by Biswa96** [comment]
Some cmake file contains full build path reference. Those should be replaced with `_IMPORT_PREFIX`. Here is the output:


```
/ucrt64/share/amgcl/cmake/amgcl-targets.cmake:  INTERFACE_INCLUDE_DIRECTORIES "D:/M/msys64/ucrt64/include;${_IMPORT_PREFIX}/include"
/ucrt64/share/amgcl/cmake/amgcl-targets.cmake:  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "D:/M/msys64/ucrt64/include"
```

**Comment by ImperatorS79** [comment]
Is there an automatic cmake way of doing this or should  use `sed` ?

**Comment by Biswa96** [comment]
Yes, some packages have that sed command e.g. hdf5, arpack etc.

**Pull request merged**